app/src/main/java/org/mozilla/fenix/home/recentvisits/controller/RecentVisitsController.kt

```diff
@@ -16,60 +16,68 @@ import mozilla.components.feature.tabs.TabsUseCases.SelectOrAddUseCase
 import org.mozilla.fenix.R
 import org.mozilla.fenix.components.metrics.Event
 import org.mozilla.fenix.components.metrics.MetricController
-import org.mozilla.fenix.historymetadata.RecentlyVisitedItem.RecentHistoryGroup
-import org.mozilla.fenix.historymetadata.RecentlyVisitedItem.RecentHistoryHighlight
-import org.mozilla.fenix.historymetadata.interactor.HistoryMetadataInteractor
 import org.mozilla.fenix.home.HomeFragmentAction
 import org.mozilla.fenix.home.HomeFragmentDirections
 import org.mozilla.fenix.home.HomeFragmentStore
+import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem.RecentHistoryGroup
+import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem.RecentHistoryHighlight
+import org.mozilla.fenix.home.recentvisits.interactor.RecentVisitsInteractor
 import org.mozilla.fenix.library.history.toHistoryMetadata
 
 /**
- * An interface that handles the view manipulation of the history metadata in the Home screen.
+ * All possible updates following user interactions with the "Recent visits" section from the Home screen.
  */
-interface HistoryMetadataController {
+interface RecentVisitsController {
 
     /**
-     * @see [HistoryMetadataInteractor.onHistoryShowAllClicked]
+     * @see [RecentVisitsInteractor.onHistoryShowAllClicked]
      */
     fun handleHistoryShowAllClicked()
 
     /**
-     * @see [HistoryMetadataInteractor.onRecentHistoryGroupClicked]
+     * @see [RecentVisitsInteractor.onRecentHistoryGroupClicked]
      */
     fun handleRecentHistoryGroupClicked(recentHistoryGroup: RecentHistoryGroup)
 
     /**
-     * @see [HistoryMetadataInteractor.onRemoveRecentHistoryGroup]
+     * @see [RecentVisitsInteractor.onRemoveRecentHistoryGroup]
      */
-    fun handleRemoveRecentHistoryGroup(searchTerm: String)
+    fun handleRemoveRecentHistoryGroup(groupTitle: String)
 
     /**
-     * @see [HistoryMetadataInteractor.onRecentHistoryHighlightClicked]
+     * @see [RecentVisitsInteractor.onRecentHistoryHighlightClicked]
      */
     fun handleRecentHistoryHighlightClicked(recentHistoryHighlight: RecentHistoryHighlight)
 }
 
 /**
- * The default implementation of [HistoryMetadataController].
+ * The default implementation of [RecentVisitsController].
  */
-class DefaultHistoryMetadataController(
+class DefaultRecentVisitsController(
     private val store: BrowserStore,
     private val homeStore: HomeFragmentStore,
     private val selectOrAddTabUseCase: SelectOrAddUseCase,
     private val navController: NavController,
     private val storage: HistoryMetadataStorage,
     private val scope: CoroutineScope,
     private val metrics: MetricController
-) : HistoryMetadataController {
+) : RecentVisitsController {
 
+    /**
+     * Shows the history fragment.
+     */
```

Comment:
I have a history of saying that we don't necessarily need a kDoc for overridden functions since they are defined in the Interface. I don't think I have a strong opinion on it, but I guess it makes it easier to just change the kDoc in only one place versus several when you need to update it.

Reply:
I tried cleaning up some of the comments for the interactor and the controller.
The interface as a general contract should not dictate the behavior. The implementation decides that for handling that action user will be navigated to the history fragment. Other implementations could have different behavior.

Another important refactoring here would've been to remove the references to the interactor from controller's interface. The controller is used by the interactor so it should not know about it.

Reply:
I think what you have proposed is good. I think what I would like to see then is that we move these descriptions to replace the @see kDocs we have in the controller interface. I would also like to see the same kind of "Callback" wording that you used in your Pocket Interactor/Controllers.